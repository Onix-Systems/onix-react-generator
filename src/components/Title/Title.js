import React from 'react';
import PropTypes from 'prop-types';
import styles from './sass/Title.module.scss';

export default function Title({ text }) {
  return (
    <h1 className={styles.appTitle}>
      {text}
    </h1>
  );
}

Title.propTypes = {
  text: PropTypes.string.isRequired
};
