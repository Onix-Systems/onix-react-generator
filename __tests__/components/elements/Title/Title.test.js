import React from 'react';
import { render } from '../../../settings/test-utils';
import Title from '../../../../src/components/Title/Title';

describe('<Title />', () => {
  test('should render Title component', () => {
    const mockTitle = 'mock title';
    const { asFragment } = render(<Title text={mockTitle} />);
    expect(asFragment(<Title text={mockTitle} />)).toMatchSnapshot();
  });
});
