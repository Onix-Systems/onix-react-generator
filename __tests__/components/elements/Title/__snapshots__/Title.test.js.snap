// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render Title component 1`] = `
<DocumentFragment>
  <h1
    class="appTitle"
  >
    mock title
  </h1>
</DocumentFragment>
`;
